### PR TITLE
chore(test): Behat: remove legacy Behat pipeline and config from OSS (MON-179159)

### PR DESCRIPTION
**Pull request by jdelpierre**
## Description

Decommission the legacy Behat E2E from the OSS `centreon` repo (part of MON-179159, the terminal Behat cleanup). Behat is no longer executed/maintained; the OSS suites are either migrated to Cypress or dead.

### Removed
- `centreon/behat.yml` + `centreon/features/` (the `Acknowledgement`, `AcknowledgementTimeout`, `DowntimeService` bootstrap contexts — no `.feature` file remained).
- `.github/workflows/behat-test.yml` (reusable Behat workflow).
- `web.yml`: the `legacy-e2e-test` job, the `centreon/behat.yml` + `centreon/features/**` change-trigger paths, and the `legacy-e2e-test` entries in the `deliver-sources` / promote `needs`.
- `.github/actions/create-jira-ticket`: `legacy-e2e-test` removed from the job-routing list.

### Out of scope (handled separately — requires composer)
Behat composer dev-dependencies in `centreon/composer.json` + `centreon-awie/composer.json` (`behat/*`, `friends-of-behat/*`, mink drivers) + their `symfony.lock` entries + autoload-dev `Centreon\Test\Behat\*` namespaces. The `composer.lock` files must be regenerated.

### Notes
- The centreon PHP lint configs (php-cs-fixer / phpstan / rector) do **not** reference `features/`, so `backend-lint` is unaffected (no dangling path).
- centreon-awie Behat files were already removed on develop (#10693); the other OSS modules (open-tickets, dsm) have no Behat.

Refs: MON-179159

### CODE REVIEW

File: .github/workflows/web.yml [hunk 1]
```diff
@@ -1293,7 +1256,7 @@ jobs:
 
   deliver-sources:
     runs-on: ubuntu-24.04
-    needs: [get-environment, e2e-test, legacy-e2e-test]
+    needs: [get-environment, e2e-test]
```
[the remark this answers is not in the document]
Comment (jdelpierre) [reply .github/workflows/web.yml:1259]: Out of scope for this PR. This change only removes the now-deleted `legacy-e2e-test` job from the `needs` (it is decommissioned together with Behat). The prior `needs` was `[get-environment, e2e-test, legacy-e2e-test]` — none of which are the cache producers (`frontend-web-build`, `backend-dependencies`, …), so the `needs`-completeness concern is pre-existing and unrelated to this Behat-decommission PR. It can be addressed separately if needed.

**Review by mushroomempires — APPROVED** [review]

**Review by a-launois — APPROVED** [review]
